cmd/network/verification.go

```diff
@@ -429,6 +439,101 @@ func (e *EgressVerification) getSubnetIds(ctx context.Context) ([]string, error)
 	// For non-PrivateLink BYOVPC clusters, provided subnets are 50/50 public/private subnets, so make the user decide for now
 	// TODO: Figure out via IGW/NAT GW/Route Tables
 	return nil, fmt.Errorf("unable to determine which non-PrivateLink BYOVPC subnets are private yet, please check manually and provide the --subnet-id flag")
+
+}
+
+// This function checks the gateway attached to the subnet and returns true if the subnet starts with igw- (for InternetGateway) and has a route to 0.0.0.0/0
+func (e *EgressVerification) isSubnetPublic(ctx context.Context, subnetID string) (bool, error) {
```

Comment:
This code seems to be similar to https://github.com/openshift/osdctl/blob/master/cmd/cluster/cpd.go#L125

Can you pull out the common logic into a separate shared function?

Reply:
I linked the other function above and that function and this function have most of the same code. I would recommend creating a third function, something called `findRouteTableForSubnet` and placing all the shared code there, then calling it within isSubnetPublic and the other fucntion I linked

Reply:
Where should I put this shared function? A new file in /pkg/utils ?  Just want to make sure I understand what the ask is.

Reply:
I am thinking /pkg/utils/utils

Reply:
I spent some time digging into this issue this morning and speaking with @mjlshen in [this slack thread](https://redhat-internal.slack.com/archives/C04PUGBE7JN/p1715951689864719)- I think the right path forward here is to leave this for now and have a future effort to clean up all of the functions in [this interface](https://github.com/openshift/osdctl/blob/f475456b95544fce30378f4a7684903e0626efb1/pkg/provider/aws/client.go#L47) such that they don't hide the `context` parameter. Then our wrapper interface will be directly compatible with base AWS clients while still allowing all of the helper functions we provide still.

Curious to hear other thoughts, as I think a refactor here would be a bit large for this PR.

Reply:
All the better - I'm also perfectly happy deprecating the wrapper interface as a whole and directly using clients from the aws go sdk :)